atomic_reactor/plugins/exit_store_metadata.py

```diff
@@ -61,50 +62,45 @@ def get_digests(self):
         """
 
         digests = {}  # repository -> digest
-        wf_data = self.workflow.data
-        for registry in wf_data.push_conf.docker_registries:
-            for image in wf_data.tag_conf.images:
-                image_str = image.to_str()
-                if image_str in registry.digests:
-                    digest = registry.digests[image_str]
-                    digests[image.to_str(registry=False)] = digest
+        registry = self.workflow.conf.registry
 
-        return digests
+        for image in self.workflow.data.tag_conf.images:
+            image_str = image.to_str()
+            image_digests = get_manifest_digests(image, registry['uri'], registry['insecure'],
+                                                 registry.get('secret', None))
+            if image_str in image_digests:
+                digest = image_digests[image_str]
+                digests[image.to_str(registry=False)] = digest
 
-    def _get_registries(self):
-        """
-        Return a list of registries that this build updated
-        """
-        return self.workflow.data.push_conf.all_registries
+        return digests
 
     def get_repositories(self):
-        tag_conf = self.workflow.data.tag_conf
-
         # usually repositories formed from NVR labels
         # these should be used for pulling and layering
         primary_repositories = []
-        for registry in self._get_registries():
-            for image in tag_conf.primary_images:
-                registry_image = image.copy()
-                registry_image.registry = registry.uri
-                primary_repositories.append(registry_image.to_str())
+
+        for image in self.workflow.data.tag_conf.primary_images:
+            registry_image = image.copy()
+            registry_image.registry = self.workflow.conf.registry['uri']
```

Comment:
I think that these two lines are not necessary, because at this point (exit plugin) all images should have the information for registry filled (It happens in pre_tag_from_config plugin)

Reply:
The same applies for other occurrences in this PR

Reply:
removed it from here and everywhere else.